tmk_core/common/report.h

```diff
@@ -157,13 +157,23 @@ typedef struct {
 } __attribute__ ((packed)) report_keyboard_t;
 */
 
+#ifdef ENABLE_16_BIT_MOUSE_REPORT
```

Comment:
@tmk I added a preprocessor option here and in the descriptor so that by default the mouse report will remain 8-bit until it's toggled to be 16-bit. The goal for this update is to make it backwards compatible so that it's not a breaking change for other projects.

Reply:
I believe that current mouse implementation support only Boot protocol(not Report protocol) and guess it won't work with some environment like UEFI, BIOS or small devices. Modern OSes won't have problem with the 16-bit report and I confirmed Linxu work well with it.

I'll look into that.

Reply:
I tested this out on some of my devices and it works on Linux 5.8.0, Windows 10 (latest), and even back to Mac OS X 10.4 (circa 2007). I did find an issue in the Open Firmware boot menu on my old Power Mac G4 which has a GUI to select the boot disk. The buttons worked, as expected, but the 16-bit values were overflowing. I added a user facing build option in `Makefile` that can be used to disable the 16-bit mouse report entirely if someone finds this to be an issue they face.

Reply:
Confirmed UEFI on my PC can recognized the extended mouse, it seems that modern boot firmware can parse HID report descriptor, surprisingly.

I'll work on extended mouse report with boot protocol support for backward compatiblity with open new issue or PR. And this ADB scroll feature will be merged once the extended mouse report is finished. This may take time.